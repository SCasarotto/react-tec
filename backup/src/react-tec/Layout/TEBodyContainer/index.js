import TEBodyContainer from './TEBodyContainer'

export default TEBodyContainer
