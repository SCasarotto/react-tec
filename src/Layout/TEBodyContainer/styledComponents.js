import styled from 'styled-components'

export const BodyContainer = styled.div`
	${(props) => {
		const { sidebarWidth } = props
		return `
            position: relative;
            width: calc(100% - ${sidebarWidth}px);
            min-height: 100%;
            margin-left: ${sidebarWidth}px;
        `
	}}
`
