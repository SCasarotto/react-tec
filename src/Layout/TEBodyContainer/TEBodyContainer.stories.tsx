import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEBodyContainer, TEBodyContainerProps } from './TEBodyContainer'

export default {
	title: 'Layout/TEBodyContainer',
	component: TEBodyContainer,
	argTypes: {},
} as Meta

const Template: Story<TEBodyContainerProps> = (args) => <TEBodyContainer {...args} />

export const Default = Template.bind({})
Default.args = {}
