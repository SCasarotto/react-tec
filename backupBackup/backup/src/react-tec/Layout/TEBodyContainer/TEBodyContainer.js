import React from 'react'

import { BodyContainer } from './styledComponents'

const TEBodyContainer = (props) => {
	const { children, ...rest } = props
	return <BodyContainer {...rest}>{children}</BodyContainer>
}

TEBodyContainer.defaultProps = {
	sidebarWidth: 200,
}

export default TEBodyContainer
