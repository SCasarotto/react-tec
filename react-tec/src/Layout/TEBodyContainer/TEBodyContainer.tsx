import React from 'react'
import PropTypes from 'prop-types'

import { BodyContainer } from './styledComponents'

const TEBodyContainer = (props) => {
	const { className = '', ...rest } = props
	return <BodyContainer className={`TEBodyContainer ${className}`} {...rest} />
}

TEBodyContainer.propTypes = {
	sidebarWidth: PropTypes.number,
}
//TODO: Not if love with this. It might be a good idea to find a better way while also introducing some default responsiveness
TEBodyContainer.defaultProps = {
	sidebarWidth: 200,
}

export default TEBodyContainer
