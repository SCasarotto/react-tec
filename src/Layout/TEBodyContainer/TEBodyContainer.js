import React from 'react'
import PropTypes from 'prop-types'

import { BodyContainer } from './styledComponents'

const TEBodyContainer = (props) => <BodyContainer {...props} />

TEBodyContainer.propTypes = {
	sidebarWidth: PropTypes.number,
}

TEBodyContainer.defaultProps = {
	sidebarWidth: 200,
}

export default TEBodyContainer
