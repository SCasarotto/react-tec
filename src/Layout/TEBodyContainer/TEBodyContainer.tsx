import React from 'react'

import { BodyContainer } from './styledComponents'

export interface TEBodyContainerProps extends React.HTMLAttributes<HTMLDivElement> {
	sidebarWidth?: number
}
/**
 *
 * `TEBodyContainer` is really just a sized meant to play well with `TESideNavbar`. Below is the exact code used in this site.
 *
 * ```
 *	<TEBodyContainer>
 *		<TETitleBar title={barTitle} />
 *		{children}
 *	</TEBodyContainer>
 * ```
 *
 */
export const TEBodyContainer: React.FC<TEBodyContainerProps> = (props) => {
	const { className = '', ...rest } = props
	return <BodyContainer className={`TEBodyContainer ${className}`} {...rest} />
}

//TODO: Not in love with this. It might be a good idea to find a better way while also introducing some default responsiveness
TEBodyContainer.defaultProps = {
	sidebarWidth: 200,
}
