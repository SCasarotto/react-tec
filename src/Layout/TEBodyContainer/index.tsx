import React from 'react'

import { BodyContainer } from './styledComponents'

export interface TEBodyContainerProps
	extends React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
	//TODO: Fix this. styled-components is weird with refs
	ref?: any
	sidebarWidth?: number
}
export const TEBodyContainer: React.FC<TEBodyContainerProps> = (props) => {
	const { className = '', ...rest } = props
	return <BodyContainer className={`TEBodyContainer ${className}`} {...rest} />
}

//TODO: Not if love with this. It might be a good idea to find a better way while also introducing some default responsiveness
TEBodyContainer.defaultProps = {
	sidebarWidth: 200,
}
