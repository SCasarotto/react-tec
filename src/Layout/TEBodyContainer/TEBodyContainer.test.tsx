import React from 'react'
import { render } from '@testing-library/react'
import { TEBodyContainer } from './TEBodyContainer'

describe('TEBodyContainer', () => {
	// Renders Without Error
	it('matches snapshot', () => {
		const { asFragment } = render(<TEBodyContainer />)
		expect(asFragment()).toMatchSnapshot()
	})

	// Class Name
	it('accepts custom className', () => {
		const { container } = render(<TEBodyContainer className='myClassName' />)
		expect(container.firstChild).toHaveClass('myClassName')
	})
})
