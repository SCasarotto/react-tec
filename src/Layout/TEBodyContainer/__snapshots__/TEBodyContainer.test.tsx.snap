// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEBodyContainer matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  width: calc(100% - 200px);
  min-height: 100vh;
  margin-left: 200px;
}

<div
    class="c0 TEBodyContainer "
  />
</DocumentFragment>
`;
